client.go

```diff
@@ -146,30 +180,27 @@ func (c *Client) ExecCmd(cmd *Cmd) ([]string, error) {
 		return nil, err
 	}
 
-	lines := make([]string, 0, 10)
-	for c.scanner.Scan() {
-		l := c.scanner.Text()
-		if l == "error id=0 msg=ok" {
-			if cmd.response != nil {
-				if err := DecodeResponse(lines, cmd.response); err != nil {
-					return nil, err
-				}
+	err := <-c.err
+	if err.Error() == "ok (0)" {
+		if cmd.response != nil {
+			if err = DecodeResponse(c.res, cmd.response); err != nil {
+				return nil, err
 			}
-			return lines, nil
-		} else if matches := respTrailerRe.FindStringSubmatch(l); len(matches) == 4 {
-			return nil, NewError(matches)
-		} else {
-			lines = append(lines, l)
 		}
+
+		return c.res, nil
 	}
 
-	return nil, c.scanErr()
+	return nil, err
 }
 
 // Close closes the connection to the server.
 func (c *Client) Close() error {
 	_, err := c.Exec("quit")
 	err2 := c.conn.Close()
+
+	close(c.notify)
```

Comment:
This is racey, the processor should close it in a defer to avoid a panic on shutdown